src/persist-client/src/internal/machine.rs

```diff
@@ -988,7 +988,7 @@ where
         self,
         reader_id: LeasedReaderId,
         gc: GarbageCollector<K, V, T, D>,
-    ) -> Vec<JoinHandle<()>> {
+    ) -> Vec<AbortOnDropHandle<()>> {
```

Comment:
Curious what @danhhz thinks, but when I last touched this code I intentionally left this as a `JoinHandle` because I feel like it's a weird API to force your caller to take an `AbortOnDropHandle` when they might not want it.

Reply:
The alternative is to move the `.abort_on_drop()` into the caller? Yeah, I think I mildly prefer that

Reply:
i can move this!